astropy/coordinates/builtin_frames/ecliptic_transforms.py

```diff
@@ -69,16 +69,20 @@ def icrs_to_helioecliptic(from_coo, to_frame):
     if not u.m.is_equivalent(from_coo.cartesian.x.unit):
         raise UnitsError(_NEED_ORIGIN_HINT.format(from_coo.__class__.__name__))
 
-    pvh, pvb = erfa.epv00(*get_jd12(to_frame.equinox, 'tdb'))
-    delta_bary_to_helio = pvh[..., 0, :] - pvb[..., 0, :]
+    # get barycentric sun coordinate
+    # this goes here to avoid circular import errors
+    from ..solar_system import get_body_barycentric
+    bary_sun_pos = get_body_barycentric('sun', to_frame.equinox)
 
-    # first offset to heliocentric
-    heliocart = (from_coo.cartesian.xyz).T + delta_bary_to_helio * u.au
+    # offset to heliocentric
+    heliocart = CartesianRepresentation(from_coo.cartesian.x + bary_sun_pos.x,
```

Comment:
Hmm, now I wonder whether to optimize when really we just want to have #5301 in... The way this is written, `cartesian` gets called three times, which each time will calculate all of `x`, `y`, and `z`, so this can be sped up nearly a factor of three by doing
```
from_coo_cartesian = from_coo.cartesian
heliocart = ... from_coo_cartesian.x + ...
```